### ISSUE TITLE
Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES]

### ISSUE THREAD

**Issue by RCoeurjoly** [koreader/koreader#6053]
When doing `adb install koreader-android-arm-v2020.03.2-33-gf756a6a_2020-04-14.apk`
For a locally generated apk

I get Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES]

Shouldn't it process the koreader-android-arm-v2020.03.2-33-gf756a6a_2020-04-14.apk.sig file?

**Comment by pazos** [comment]
Nope, release builds are unsigned. Debug builds (using `--debug` flag) are signed with test keys.

It is intended as both our repo and f-droid expect an unsigned package to sign it later. There's a beautiful message at the end of the build

```
Application KOReader was built, type: release (unsigned),       flavor: rocks, version: v2020.03.2-32-gec5b569a_2020-04-14, release 7397
WARNING: You'll need to sign this application to be able to install it
```

**Comment by RCoeurjoly** [comment]
> Nope, release builds are unsigned. Debug builds (using `--debug` flag) are signed with test keys.
> 
> It is intended as both our repo and f-droid expect an unsigned package to sign it later. There's a beautiful message at the end of the build
> 
> ```
> Application KOReader was built, type: release (unsigned),       flavor: rocks, version: v2020.03.2-32-gec5b569a_2020-04-14, release 7397
> WARNING: You'll need to sign this application to be able to install it
> ```

The right command is:

`./kodev release --debug android`

`./kodev release android --debug ` doesn't sign the apk

**Comment by pazos** [comment]
@RCoeurjoly: Yeah, that's true. Both flags `--no-translation` and `--debug` go before the target.

BTW, I see you modified the wiki to add instructions of how to obtain a signed build. Thanks for helping but a debug build is more/different than a release/signed build. It is twice (or more) slow. Takes 3-4x to uncompress the first time after install, it has debug symbols so the apk weights more.

Finally, as debug builds are signed with the same key anybody can build a package with the same id and the program will install on top of the other one, so security go poof.

The right way of building a signed package is build a release package and sign it with your own key. Keep in mind that you won't be able to update the package with other releases (from github or f-droid) without uninstalling your package first, as signatures do not match.

TL;DR: Unless you're developing something android specific, I wouldn't recommend building/signing your own updates. You will need to do that again for each release.
